### ISSUE TITLE
BadRequest URLs should serve a nicer error page

### ISSUE THREAD

**Issue by jameshadfield** [nextstrain/nextstrain.org#774]
### Current Behavior

URLs which result in `BadRequest` errors server side send an overly plain HTML page with "Bad Request" (in production mode) or the stack trace (in dev mode). 

As an example, the url path `/staging/WNV_NA` (which is a bad request because underscores aren't allowed for datasets) looks like so (production nextstrain.org on the left,  localhost on the right):
<img width="1344" alt="image" src="https://github.com/nextstrain/nextstrain.org/assets/8350992/dd57eb41-ba48-4a29-b6a7-93cd8fd667e6">

### Expected behavior
A page more like that for a valid but non-existing (404) dataset:

<img width="1064" alt="image" src="https://github.com/nextstrain/nextstrain.org/assets/8350992/a62b08f6-1850-4512-bfa4-2bb0b8936395">

### Your environment: if browsing Nextstrain online
Nextstrain.org production server running from 24ba9ee7

### Additional context
I haven't looked into it yet but I believe Express automatically handles NotFound errors, which may be the reason they don't result in similar behaviour.

**Comment by jameshadfield** [comment]
As an extension of this issue, and one way to solve it:

We sometimes include helpful information in the various HTTP Errors (server-side), such as
```
NotFound(`Version descriptor ${versionDescriptor} predates available versions`)
BadRequest(`Requested version must be in YYYY-MM-DD format (version descriptor requested: "${versionDescriptor}")`)
BadRequest(`Resource (e.g. dataset and narrative) paths may not include underscores (_); use slashes (/) instead`);
```

Were the returned Gatsby page able to access this information (and the corresponding error code) we could then communicate this which would be a nice improvement over the current state.

**Comment by corneliusroemer** [comment]
Bump for this issue, just encountered it and had to look at console to figure out what was going on. Less tech-savvy users would not be able to figure this out.

<img width="1035" alt="image" src="https://github.com/user-attachments/assets/1d01de41-5986-4a5f-8c05-4d4e8d5a9bac" />

**Comment by tsibley** [comment]
@corneliusroemer I don't think that error you screenshotted fundamentally has anything to do this with issue. You only ran into a BadRequest by circumstance there, and it was only visible in the dev console. This issue is about them being very user-visible in the viewport, as in James screenshot above.

The root issue for your example—as I'm sure you know, but to make it clear for anyone else reading—was the inclusion of `https://` in the URL path after `…/fetch/`. Auspice interpreted that as a tanglegram request for two (bad) datasets, `fetch/https` and `raw.githubusercontent.com/…`, instead of a request for a single (good) dataset, `fetch/raw.githubusercontent.com/…`.

Now, I agree 100% that most users would have a hard time figuring that out, but I think improving the error case there is very separate from this issue.

**Comment by victorlin** [comment]
We're now using Next.js instead of Gatsby, but the limitations are similar. The main limitation is that Next.js pages are only capable of returning 200, 404, or 403 (see [docs](https://nextjs.org/docs/app/api-reference/file-conventions/loading#status-codes), [discussion](https://github.com/vercel/next.js/discussions/53225)).

Specifically for this issue, there's no way to handle a BadRequest error with all of the following:

- Navbar/footer/etc. from Next.js components
- Custom error message
- Proper HTTP status code

An alternative approach is to create a simple HTML template with basic branding, served directly by the Express router. This would support custom error message and proper HTTP status code.
